dev/PushNotifications/PushNotificationBackgroundTask.cpp

```diff
@@ -0,0 +1,27 @@
+﻿#pragma once
+
+#include "pch.h"
+
+#include "PushNotificationBackgroundTask.h"
+
+#include <winrt/base.h>
+#include "PushNotificationReceivedEventArgs.h"
+#include "externs.h"
+
+namespace winrt
+{
+    using namespace Windows::ApplicationModel::Background;
+    using namespace Microsoft::Windows::PushNotifications;
+}
+
+wil::unique_handle g_waitHandleForArgs;
+winrt::PushNotificationReceivedEventArgs g_activatedEventArgs{ nullptr };
+
+void PushNotificationBackgroundTask::Run(winrt::IBackgroundTaskInstance const& taskInstance)
```

Comment:
I think the first notification is processed and the second one is lost.

@pmpurifoy Did we test this scenario?

Reply:
This is a good unit test to add. I would think that both the calls here would map to the same instance of PushNotificationBackgroundTask since CoREgisterClassObject is called only once on process launch and it's flag is set to Single_Use